autoload/lsp/internal/work_done_progress.vim

```diff
@@ -0,0 +1,62 @@
+" https://microsoft.github.io/language-server-protocol/specifications/specification-current/#progress
+
+let s:progress_ui = []
+let s:enabled = 0
+
+function! lsp#internal#work_done_progress#_enable() abort
+    if !g:lsp_work_done_progress_enabled | return | endif
+
+    if s:enabled | return | endif
+    let s:enabled = 1
+
+    let s:Dispose = lsp#callbag#pipe(
+          \ lsp#stream(),
+          \ lsp#callbag#filter({x->has_key(x, 'response') && has_key(x['response'], 'method')
+          \  && x['response']['method'] ==# '$/progress' && has_key(x['response'], 'params')
+          \  && has_key(x['response']['params'], 'value') && type(x['response']['params']['value']) == type({})}),
+          \  lsp#callbag#subscribe({'next': {x->s:handle_work_done_progress(x['server'], x['response'])}})
+          \ )
+endfunction
+
+function! s:handle_work_done_progress(server, progress) abort
```

Comment:
can you rename `progress` as `response`. 

This is one thing i have realized after working in vim script codebase and vim-lsp quite a lot is that in the future it becomes quite difficult to know what the `progress` is since there is no typings. trying to use generic term helps here.

Reply:
Thank you for advice.
I renamed it.